### ISSUE TITLE
compound keys

### ISSUE THREAD

**Issue by nogece** [indexeddbshim/IndexedDBShim#54]
Let me first of all state that this project is a great! I'd love to contribute, yet, before, I know enough to do that, I can only contribute by bug reports:

A rather small but existing issue : the spec and most browsers (chrome/moz, IE10 not yet) allow compound keys:
http://stackoverflow.com/questions/14283257/indexeddb-compound-key-ie10
IndexedDBShim doesn't.

**Comment by Burstaholic** [comment]
We just spent a great deal of time tracing down this problem when trying to use this shim with compound indexes. This is an important feature for many use cases, please consider investigating a fix.

**Comment by axemclion** [comment]
Implementing Compound keys is a little hard and requires re-doing the way keys are currently done in the project. For now, I just lazily implemented keys as indexes. With compound keys, you would ned to explode the indexes, and do something more. Will look at it when I have time. It would be great if someone else could send a pull request :)

**Comment by Morgul** [comment]
In investigating this issue, I managed to figure out how to get compound keys to store properly, and I pushed my changes to a fork:

https://github.com/Morgul/IndexedDBShim/commit/8ec47afc1346abde85c98bc7cbf9f10042fb7759#diff-a6eef872c115879cee276e9fe31786e4R1134

This isn't complete by any means (it appears to store correctly, but I suspect there's more work on the retrieving side that would need to be done). It may not even be an appropriate solution, I'm not very familiar with the inner workings of this shim.

@axemclion If you have any thoughts on my work so far, especially pointers at what additionally may need to be done to get this working, I'd love to hear them. For now, I'm afraid I need to look for some other solution for our current app. 

This code can be used as the base of a pull request, if someone wants to keep going with this.
